src/irr/asset/CGLTFLoader.h

```diff
@@ -96,6 +101,27 @@ namespace irr
 					}
 				};
 
+				/*
+					A set of primitives to be rendered. A node can contain one mesh. 
+					A node's transform places the mesh in the scene.
+				*/
+
+				struct SGLTFMesh
+				{
+					struct SPrimitive
+					{
+						std::unordered_map<std::string, uint32_t> attributes;	//!< A dictionary object, where each key corresponds to mesh attribute semantic and each value is the index of the accessor containing attribute's data.
+						std::optional<uint32_t> indices;						//!< The index of the accessor that contains the indices.
+						std::optional<uint32_t> material;						//!< The index of the material to apply to this primitive when rendering.
+						std::optional<uint32_t> mode;							//!< The type of primitives to render.
+						std::optional<std::unordered_map<std::string, uint32_t>> targets;	//!< An array of Morph Targets, each Morph Target is a dictionary mapping attributes (only POSITION, NORMAL, and TANGENT supported) to their deviations in the Morph Target.
+					};
+
+					std::vector<SPrimitive> primitives;					//!< An array of primitives, each defining geometry to be rendered with a material.
+					std::optional<std::vector<uint32_t>> weights;		//!< Array of weights to be applied to the Morph Targets.
+					std::optional<std::string> name;					//!< The user-defined name of this object.
```

Comment:
ignore morph targets for now.

its important to parse this weird data structure out into separate `IMeshBuffer` (node is an `IMesh`) which only contain primitives with the same "material" (textures, and stuff), that said I'm not saying "change the structure" its fine for reading in glTF.

Reply:
yeah, I will ignore more than Morphg targets probably, but I will define everything in the glTF structs for later purposes in the future